lib/3scale_toolbox/commands/import_command/openapi/update_policies_step.rb

```diff
@@ -0,0 +1,92 @@
+module ThreeScaleToolbox
+  module Commands
+    module ImportCommand
+      module OpenAPI
+        class UpdatePoliciesStep
+          include Step
+
+          ##
+          # Updates Proxy Policies config
+          def call
+            # to be idempotent, we must read first
+            source_policies_settings = service.policies
+            # shallow copy
+            # to detect policies_settings changes, should only be updated setting objects
+            # do not update in-place, otherwise changes will not be detected
+            policies_settings = source_policies_settings.dup
+
+            add_anonymous_access_policy(policies_settings)
+            add_rh_sso_keycloak_role_check_policy(policies_settings)
+
+            return if same_policy_settings?(source_policies_settings, policies_settings)
+
+            service.update_policies('policies_config' => policies_settings)
+          end
+
+          private
+
+          def add_anonymous_access_policy(policies)
+            # only on 'open api' security req
+            return unless security.nil?
+
+            def_credentials = policies.find { |policy| policy['name'] == 'default_credentials' }
+
+            return unless def_credentials.nil?
+
+            # Anonymous policy should be before apicast policy
+            # hence, adding as a first element
+            policies.insert(0, anonymous_policy)
+          end
+
+          def anonymous_policy
+            raise ThreeScaleToolbox::Error, 'User key must be provided by ' \
+              '--default-credentials-userkey optional param' \
+              if default_credentials_userkey.nil?
+
+            {
+              'name': 'default_credentials',
+              'version': 'builtin',
+              'configuration': {
+                'auth_type': 'user_key',
+                'user_key': default_credentials_userkey
+              },
+              'enabled': true
+            }
+          end
+
+          def add_rh_sso_keycloak_role_check_policy(policies)
+            # only applies to oauth2 sec type
+            return if security.nil? || security.type != 'oauth2'
+
+            keycloak_policy_idx = policies.find_index { |policy| policy['name'] == 'keycloak_role_check' }
+
+            return unless keycloak_policy_idx.nil?
```

Comment:
why find_index ? isn't `.any?` enough?

Reply:
yeah, you are right. Long history short is legacy code from old implementation.